client/luigi-element.d.ts

```diff
@@ -0,0 +1,411 @@
+   * @memberof linkManager
+   * @param {string} path path which existence you want to check
+   * @returns {promise} a promise which resolves to a Boolean variable specifying whether the path exists or not
+   * @example
+   *  let pathExists;
+   *  LuigiClient
+   *  .linkManager()
+   *  .pathExists('projects/pr2')
+   *  .then(
+   *    (pathExists) => {  }
+   *  );
+   */
+  pathExists: (path: string) => Promise<boolean>;
+
+  /**
+   * Discards the active view and navigates back to the last visited view. Works with preserved views, and also acts as the substitute of the browser **back** button. **goBackContext** is only available when using preserved views.
+   * @memberof linkManager
+   * @param {any} goBackValue data that is passed in the **goBackContext** field to the last visited view when using preserved views
+   * @example
+   * LuigiClient.linkManager().goBack({ foo: 'bar' });
+   * LuigiClient.linkManager().goBack(true);
+   */
+  goBack: (goBackValue: any) => this;
+
+  /**
+   * Checks if there is one or more preserved views. You can use it to show a **back** button.
+   * @memberof linkManager
+   * @returns {boolean} indicating if there is a preserved view you can return to
+   */
+  hasBack: () => boolean;
+}
+
+export declare class LuigiElement extends HTMLElement {
+  constructor(options?: Options);
+  /**
+   * Override to return the html template string defining the web component view.
+   *
+   * @param {*} ctx The context object passed by luigi core
+   */
+  render(ctx?: Object): string;
+  /**
+   * Override to execute logic after an attribute of this web component has changed.
+   */
+  update(): void;
+  /**
+   * Override to execute logic when a new context object is set.
+   *
+   * @param {*} ctx The new context object passed by luigi core
+   */
+  onContextUpdate(ctx: Object): void;
+  /**
+   * Override to execute logic after initialization of the web component, i.e.
+   * after internal rendering and all context data set.
+   *
+   * @param {*} ctx The context object passed by luigi core
+   */
+  afterInit(ctx: Object): void;
+
+  /**
+   * Query selector operating on shadow root.
+   *
+   * @see ParentNode.querySelector
+   */
+  querySelector(selector: string): any;
+
+  /**
+   * LuigiClient instance
+   */
+  LuigiClient: LuigiClient;
+
+  /**
+   * Context object
+   * @returns {Object} context object
+   */
+  get context(): Object;
+}
+
+export declare interface Options {
+  /**
+   *
+   *if `true` shadowRoot mode is "open" otherwise shadowRoot mode is "closed".
+   */
+  openShadow: boolean;
+}
+
+/**
+ * Html string processing according to luigi functionality.
+ * Also useful in combination with LitElement VS Code plugins.
+ *
+ * @param {String} literal The literal to process.
+ * @returns {String} Returns the processed literal.
+ */
+export declare const html: (strings: TemplateStringsArray, ...values: unknown[]) => string;
```

Comment:
(question) In the luigi-element.js this method receives ```keys``` [luigi-element.js: L148](https://github.com/SAP/luigi/blob/master/client/src/luigi-element.js#L148). But it's declared as values here. Should it be like this? Maybe I'm overeager but there's no param for this as well. I presume it's all by design, if so, ignore my comment. :)